src/components/WriteEditModal/ImageInput.tsx

```diff
@@ -0,0 +1,58 @@
+'use client';
+
+import React, { useState } from 'react';
+import Image from 'next/image';
+import Carmera from '@/public/svgs/camera.svg';
+import DeleteImage from '@/public/svgs/deleteImage.svg';
+import classNames from 'classnames/bind';
+import styles from './ImageInput.module.scss';
+
+export default function ImageInput() {
+  const cn = classNames.bind(styles);
+
+  const [selectedImageFile, setSelectedImageFile] = useState<string[]>([]);
+  const handleChangeImage = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (selectedImageFile.length === 4) {
+      alert('이미지 4개 이상 불가');
+      return;
+    }
+    if (e.target.files) {
+      const file = e.target.files[0];
+      const imageUrl = URL.createObjectURL(file);
+      setSelectedImageFile([...selectedImageFile, imageUrl]);
+    }
+  };
+  const handleClickDeleteImage = (clickedImageIndex: number) => {
+    const updatedImageFile = [
+      ...selectedImageFile.slice(0, clickedImageIndex),
+      ...selectedImageFile.slice(clickedImageIndex + 1),
+    ];
+    setSelectedImageFile(updatedImageFile);
+  };
```

Comment:
statae 변경할 때, 이전 값을 참조할 때는 직접 참조하는 것보다 setter함수의 prev를 이용하는 게 더 안전한 방법입니다.
```suggestion
  const handleClickDeleteImage = (clickedImageIndex: number) => {
    setSelectedImageFile((prev) => [...prev.slice(0, clickedImageIndex), ...prev.slice(clickedImageIndex + 1)]);
  };
```